runtime/interpreter/decode.go

```diff
@@ -943,7 +943,23 @@ func (d StorableDecoder) decodeStorageCapability() (*StorageCapabilityValue, err
 		return nil, errors.NewUnexpectedError("invalid capability borrow type encoding: %w", err)
 	}
 
-	return NewStorageCapabilityValue(d.memoryGauge, address, pathValue, borrowType), nil
+	// Decode ID at array index encodedStorageCapabilityValueIDFieldKey
+
+	id, err := d.decoder.DecodeUint64()
```

Comment:
We would need to update CCF accordingly I guess. Maybe create an issue

Reply:
Good point! Created #2432 to track that